crates/parser/src/grammar/generic_args.rs

```diff
@@ -72,28 +72,24 @@ fn lifetime_arg(p: &mut Parser) {
     m.complete(p, LIFETIME_ARG);
 }
 
-// test const_arg
-// type T = S<92>;
-pub(super) fn const_arg(p: &mut Parser) {
-    let m = p.start();
```

Comment:
This is not right, this should parse a `CONST_ARG`. You shouldn't be touching this function at all, the problem I mentioned on zulip was the call to this function itself. `const_param` should not call `const_arg`, as it doesn't want to parse a `CONST_ARG`, it wants to parse an expression.

Reply:
I agree, but const_arg still parses a CONST_ARG and const_param no longer calls const_arg. The logic needed by const_param is the same as the inner logic of const_arg which is why it's now shared.

Reply:
Oh sorry, I didn't see you putting the new declaration below, I expected `const_arg` to stay above hence why i missed it.
And yes apparently you are right, the parsing is actually the same for both as is done by rustc.

I'd say we should rename the function to something like `const_arg_expr` though, I am not a fan of the word `content`.